apps/state/test/state/route_pattern_test.exs

```diff
@@ -47,6 +47,11 @@ defmodule State.RoutePatternTest do
       assert filter_by(%{route_ids: ["route"], direction_id: 1}) == []
       assert filter_by(%{route_ids: ["route"], stop_ids: ["stop"]}) == [route_pattern]
       assert filter_by(%{route_ids: ["not_route"], stop_ids: ["stop"]}) == []
+      assert filter_by(%{canonical: true}) == [route_pattern]
+      assert filter_by(%{canonical: false}) == []
+      assert filter_by(%{canonical: nil}) == [route_pattern]
```

Comment:
suggestion: this doesn't seem valid, as the `RoutePattern.filters()` type stays that `canonical` is a `boolean`.